Data/ByteString/Short/Internal.hs

```diff
@@ -446,6 +554,781 @@ concat sbss =
       copyByteArray (asBA src) 0 dst off len
       copy dst (off + len) sbss
 
+-- ---------------------------------------------------------------------
+-- Basic interface
+
+infixr 5 `cons` --same as list (:)
+infixl 5 `snoc`
+
+-- | /O(n)/ Append a byte to the end of a 'ShortByteString'
+-- 
+-- Note: copies the entire byte array
+--
+-- @since 0.11.3.0
+snoc :: ShortByteString -> Word8 -> ShortByteString
+snoc = \sbs c -> let l = length sbs
+                     nl = l + 1
+  in create nl $ \mba -> do
+      copyByteArray (asBA sbs) 0 mba 0 l
+      writeWord8Array mba l c
+{-# INLINE snoc #-}
+
+-- | /O(n)/ 'cons' is analogous to (:) for lists.
+--
+-- Note: copies the entire byte array
+--
+-- @since 0.11.3.0
+cons :: Word8 -> ShortByteString -> ShortByteString
+cons c = \sbs -> let l = length sbs
+                     nl = l + 1
+  in create nl $ \mba -> do
+      writeWord8Array mba 0 c
+      copyByteArray (asBA sbs) 0 mba 1 l
+{-# INLINE cons #-}
+
+-- | /O(1)/ Extract the last element of a ShortByteString, which must be finite and non-empty.
+-- An exception will be thrown in the case of an empty ShortByteString.
+--
+-- @since 0.11.3.0
+last :: ShortByteString -> Word8
```

Comment:
I would rather do not introduce new partial API in 2022 and rather push users to `unsnoc` / `uncons`. @sjakobi what do you think?

Reply:
Well, I definitely need these functions for AFPP and I'd rather not redefine them or create obscure workarounds.

Reply:
Aren't `unsnoc` / `uncons` unsufficient for AFPP?

Reply:
I think there are cases where using partial functions like `last` makes sense.

Also, we already offer `last` for strict and lazy `ByteString`, and I suspect that we'll eventually want to offer roughly the same API for `ShortByteString`.

```suggestion
last :: HasCallStack => ShortByteString -> Word8
```

Reply:
> Aren't unsnoc / uncons sufficient for AFPP?

I can even do without `unsnoc`/`uncons` and re-define them on my own. Does that mean we should get rid of them?

I don't buy into the "no partial functions in Haskell" discussion. Haskell is not a total language and will never be. GHC is too "dumb" to understand that something like this is total:

```hs
foo :: ShortByteString -> ShortByteString
foo sbs
  | null sbs = sbs
  | last sbs == _space = "lol"
  | otherwise = "shrug"
```

This is a common pattern and I see now reason to force people to jump through hoops to write this code.

Reply:
All right. Please a) annotate partial functions with `HasCallStack`, b) add `uncons` and `unsnoc`.